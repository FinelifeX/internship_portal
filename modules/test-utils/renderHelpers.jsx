/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import TR from 'react-test-renderer';
import { render as RTLRender } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

/**
 * @typedef {Object.<string, any>} Props
 */

/**
 * @typedef {React.ReactNode} Children
 */

/**
 * @typedef {{props?: Props, children?: Children}} RenderOptions
 */

/**
 *
 * @param {React.ComponentType} Component
 * @param {RenderOptions} [options]
 * @returns {JSX.Element}
 */
export const render = (Component, options = {}) => {
  const { props, children } = options;
  return <Component {...props}>{children}</Component>;
};

/**
 * Render React component inside MemoryRouter
 * @param {React.ComponentType} Component
 * @param {RenderOptions} [options]
 * @param {import('react-router-dom').MemoryRouterProps} routerOptions
 * @returns {JSX.Element}
 */
export const renderWithRouter = (
  Component,
  options = {},
  routerOptions = {},
) => {
  const { props, children } = options;

  return (
    <MemoryRouter {...routerOptions}>
      <Component {...props}>{children}</Component>
    </MemoryRouter>
  );
};

/**
 * Shallow render component using "react-test-renderer"
 * @type {import('react-test-renderer').create}
 */
export const shallow = TR.create;

/**
 * Render component using "render()" from "@testing-library/react"
 * @type {import('@testing-library/react').render}
 */
export const mount = RTLRender;
