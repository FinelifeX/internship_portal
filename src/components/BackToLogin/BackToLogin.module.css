.text {
  width: 100%;
  color: #818181;
}
