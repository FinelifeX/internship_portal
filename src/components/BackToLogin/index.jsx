import React from 'react';
import { Link } from 'react-router-dom';
import { AuthRoutes } from '../../routes/auth';
import styles from './BackToLogin.module.css';

const BackToLogin = () => (
  <p className={styles.text}>
    Remembered your password?{' '}
    <Link data-testid="login" to={AuthRoutes.Login}>
      Back to Log In
    </Link>
  </p>
);

export default BackToLogin;
