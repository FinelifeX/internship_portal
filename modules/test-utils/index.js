import * as RenderHelpers from './renderHelpers';

export { RenderHelpers };
