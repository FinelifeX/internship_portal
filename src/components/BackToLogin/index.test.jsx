import { shallow, renderWithRouter, mount } from 'test-utils/renderHelpers';
import BackToLogin from './index';

describe('BackToLogin component', () => {
  const TEST_ID_LINK = 'login';

  it('should render correctly', () => {
    const tree = shallow(renderWithRouter(BackToLogin)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should have link to Login page', () => {
    const { getByTestId } = mount(renderWithRouter(BackToLogin));

    expect(getByTestId(TEST_ID_LINK)).toBeInTheDocument();
  });
});
