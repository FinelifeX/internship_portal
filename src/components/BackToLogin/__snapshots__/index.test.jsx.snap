// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackToLogin component should render correctly 1`] = `
<p
  className="text"
>
  Remembered your password?
   
  <a
    data-testid="login"
    href="/auth/login"
    onClick={[Function]}
  >
    Back to Log In
  </a>
</p>
`;
